operator-common/src/main/java/io/strimzi/operator/common/operator/resource/AbstractResourceOperator.java

```diff
@@ -240,6 +240,9 @@ public T get(String namespace, String name) {
      * @return A Future for the result.
      */
     public Future<T> getAsync(String namespace, String name) {
+        if (name == null || name.isEmpty()) {
+            throw new IllegalArgumentException(resourceKind + " with an empty name cannot be configured. Please provide a name.");
```

Comment:
What about in the synchronous `get` above? And in the `AbstractNonNamespacedResourceOperator`?